### ISSUE TITLE
sbt client (--client) does not print error output on repeated commands, just quits with "[error]..."

### ISSUE THREAD

**Issue by simlei** [sbt/sbt#6698]
## steps

- Setup minimal project with sbt `1.5.5`
- with some main method that fails, lets say at runtime (but also happens sometimes with compiletime error output)
- have the sbt server powered down
- run `sbt --client runMain <your Main class>` and observe error output
- run the same command again -- **and just get the message that there was an error, but no details**
- run `sbt --client shutdown`, and repeat: the first time -- when the server boots up -- the error message is there, the second time run, it is not.

## problem

Reproducible on my machine only on recurring runs of `sbt --client` as described above, the thin client **sometimes doesn't produce any error output on runtime / compiletime errors**. Sometimes it works (on a running sbt instance) and then stops working later.

Example output: -- first time, error gets printed, second time, it doesn't get printed.

```
should print
[error] (run-main-0) java.lang.AssertionError: assertion failed
[error]
[error] obj1 == obj2
[error] |    |  |
[error] |    |  IS(4,blub)
[error] |    false
[error] IS(3,hellooooo)
[error] java.lang.AssertionError: assertion failed
[error]
[error] obj1 == obj2
[error] |    |  |
[error] |    |  IS(4,blub)
[error] |    false
[error] IS(3,hellooooo)
[error]
[error]         at com.eed3si9n.expecty.ExpectyBase$ExpectyListener.expressionRecorded(Expecty.scala:41)
[error]         at com.eed3si9n.expecty.RecorderRuntime.recordExpression(RecorderRuntime.scala:40)
[error]         at bci.logic.Stuff$.StuffMain(Stuff.scala:21)
[error]         at bci.logic.StuffMain.main(Stuff.scala:7)
[error] Total time: 1 s, completed Oct 21, 2021 6:18:05 PM
snuc@snuc-NUC8i7BEH:~/sandbox/bct/bci-logic$ sbt --client runMain bci.logic.StuffMain
[info] entering *experimental* thin client - BEEP WHIRR
[info] terminate the server with `shutdown`
> runMain bci.logic.StuffMain
[error] Total time: 0 s, completed Oct 21, 2021 6:18:07 PM
snuc@snuc-NUC8i7BEH:~/sandbox/bct/bci-logic$
```

## expectation

it should just print errors every time.

## notes

Maybe the client pipe gets closed too early or not flushed?

**Comment by eed3si9n** [comment]
Thanks for the report!

**Comment by simlei** [comment]
sbt version `1.5.2` seems to work better, i.e. did not get this behavior again with this version in `project/build.properties`

**Comment by simlei** [comment]
Noticed with joy people were working on this. However, using sbt 1.5.7, the problem is not solved for me.
- In my case, I am not shown the problem of a build-level failure. 
- It is not, as the issue title would suggest, a problem of repeated running of sbtn — the server was shutdown before. running sbt --client compile the acquits with just `error`. 
- If I run sbt without client, I get the full problem (a stacktrace and stuff explaining that there are conflicting cross-versions in my case) which is what I want.

### actual output with --client (problematic)
```
[[0minfo[0m] entering *experimental* thin client - BEEP WHIRR
[[0minfo[0m] server was not detected. starting an instance
[info] welcome to sbt 1.5.7 (AdoptOpenJDK Java 11.0.11)
>>> SNIP [...]
[info] set current project to bci_logic (in build file:/home/snuc/sandbox/bct/bci-logic/)
[info] sbt server started at local:///home/snuc/.sbt/1.0/server/e2cdec80be7c0501b2a2/sock
[[0minfo[0m] terminate the server with `shutdown`
> reload
[[[31merror[0m] sbt server disconnected
[0J
```

### Output without --client (as expected)
```
[info] welcome to sbt 1.5.7 (AdoptOpenJDK Java 11.0.11)
[info] loading global plugins from /home/snuc/.sbt/1.0/plugins
>>> SNIP [...]
[info] set current project to bci_logic (in build file:/home/snuc/sandbox/bct/bci-logic/)
[info] Executing in batch mode. For better performance use sbt's shell
[error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:/home/snuc/sandbox/bct/bci-
logic/"), "root"):
[error]    org.typelevel:cats-effect-std _3, _2.13
[error]    org.scodec:scodec-bits _2.13, _3
[error]    co.fs2:fs2-core _2.13, _3
[error]    org.typelevel:cats-effect _3, _2.13
[error]    org.typelevel:simulacrum-scalafix-annotations _3, _2.13
[error]    org.typelevel:cats-kernel _3, _2.13
[error]    org.typelevel:cats-effect-kernel _3, _2.13
[error]    org.typelevel:cats-core _3, _2.13
[error] java.lang.RuntimeException: Conflicting cross-version suffixes in: org.typelevel:cats-effect-std, org.scodec:scod
ec-bits, co.fs2:fs2-core, org.typelevel:cats-effect, org.typelevel:simulacrum-scalafix-annotations, org.typelevel:cats-ke
rnel, org.typelevel:cats-effect-kernel, org.typelevel:cats-core
[error]         at scala.sys.package$.error(package.scala:30)
[error]         at sbt.librarymanagement.ConflictWarning$.processCrossVersioned(ConflictWarning.scala:39)
[error]         at sbt.librarymanagement.ConflictWarning$.apply(ConflictWarning.scala:19)
[error]         at sbt.Classpaths$.$anonfun$ivyBaseSettings$71(Defaults.scala:3194)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error]         at java.base/java.lang.Thread.run(Thread.java:829)
[error] (update) Conflicting cross-version suffixes in: org.typelevel:cats-effect-std, org.scodec:scodec-bits, co.fs2:fs2
-core, org.typelevel:cats-effect, org.typelevel:simulacrum-scalafix-annotations, org.typelevel:cats-kernel, org.typelevel
:cats-effect-kernel, org.typelevel:cats-core
[error] Total time: 0 s, completed Jan 10, 2022, 7:11:45 PM
```

seems like I will have to fall back to 1.5.2 again — that version works for me.
Please let me know how to provide further info so that this can get solved :) Thanks for the work already done!

**Comment by SethTisue** [comment]
>  sbt 1.5.7, the problem is not solved for me

how about in 1.6.1?

**Comment by adpi2** [comment]
This fix was not backported to 1.5.7 but it should work in the 1.6.x series.
